winforms/src/toga_winforms/widgets/base.py

```diff
@@ -15,9 +18,39 @@
 
 class Scalable:
     SCALE_DEFAULT_ROUNDING = ROUND_HALF_EVEN
+    _dpi_scale = None
 
-    def init_scale(self, native):
-        self.dpi_scale = native.CreateGraphics().DpiX / 96
+    @property
+    def dpi_scale(self):
+        if Scalable._dpi_scale is None:
+            self.update_scale()
+        return Scalable._dpi_scale
+
+    @dpi_scale.setter
+    def dpi_scale(self, value):
+        Scalable._dpi_scale = value
+
+    def update_scale(self, screen=None):
+        # Doing screen=Screen.PrimaryScreen in method signature will make
+        # the app to become unresponsive when DPI settings are changed.
+        if screen is None:
+            screen = Screen.PrimaryScreen
```

Comment:
Shouldn't we always use the screen that the window or widget  is actually on?

Reply:
I have modified it. Thanks!